src/push-state-tree.js

```diff
@@ -56,6 +51,59 @@ const CHANGE = 'change';
 const MATCH = 'match';
 const OLD_MATCH = 'oldMatch';
 
+// Internal history keep tracking of all changes in the location history, it can be reset any
+let internalHistory;
+function InternalLocation(id, url) {
+  this.id = id;
+  this.url = url;
+}
+function InternalHistory() {
+  if (!(this instanceof InternalHistory)) {
+    return internalHistory = new InternalHistory();
+  }
+
+  // It's a object like an array, but it can start from a certain point to allow memory cleanup
+  if (internalHistory && internalHistory.length) {
```

Comment:
not sure if this is what you intended, but `if (internalHistory.length)` evals to false if `.length` is 0